--- gcc-5.3.0/libstdc++-v3/include/std/thread	2016-12-11 16:51:54
+++ gcc-5.3.0/libstdc++-v3/include/std/thread.new	2023-10-10 11:32:17
@@ -79,13 +79,13 @@
       friend class thread;
       friend class hash<thread::id>;
 
-      friend bool
-      operator==(thread::id __x, thread::id __y) noexcept
-      { return __gthread_equal(__x._M_thread, __y._M_thread); }
+      // friend bool
+      // operator==(thread::id __x, thread::id __y) noexcept
+      // { return __gthread_equal(__x._M_thread, __y._M_thread); }
 
-      friend bool
-      operator<(thread::id __x, thread::id __y) noexcept
-      { return __x._M_thread < __y._M_thread; }
+      // friend bool
+      // operator<(thread::id __x, thread::id __y) noexcept
+      // { return __x._M_thread < __y._M_thread; }
 
       template<class _CharT, class _Traits>
 	friend basic_ostream<_CharT, _Traits>&
